### ISSUE TITLE
UBS related new Bug

### ISSUE THREAD

**Issue by al-jabirr** [leocg/Hackintosh-Dell-G5-5590#25]
I think this is a new problem, so felt compelled to post. When I boot the computer, if there is a device plugged into the USB port, Then it shows after the macOS boot. **But when the device is unplugged and replugged the device no longer shows or detects.**
![Screenshot 2022-12-01 at 9 46 51 AM](https://user-images.githubusercontent.com/8538407/204962726-e1dcceed-c9f8-4667-94c0-61581055e399.png)
![Screenshot 2022-12-01 at 9 56 45 AM](https://user-images.githubusercontent.com/8538407/204962738-0fd17cf3-4d98-4ff9-bbf5-ed61a99054b3.png)

**Comment by leocg** [comment]
Maybe it's something related to USB mapping. Let's try to detect the issue:

Open [Hackintool](https://github.com/benbaker76/Hackintool), click on USB. Plug the device and verify the column named 'Device' if it it displayed there. If the device don't show up, try something that requires less energy, like a pen drive. If it shows up, probably the port is not passing the energy needed to power up the device. In this case you can create a new USBMap.kext and replace the original under EFI/OC/Kexts. If your USB controller is different, you may need to remap the ports to work correctly. 

You can see if the port is working as expected if you plug an iPhone or iPad and see the information "Extra Operating Current (mA):	1600" in System Profiler -> USB:

![Screenshot 2022-12-02 at 11 18 12](https://user-images.githubusercontent.com/3638669/205313643-31e43c5b-13df-4ca4-a507-48be82d8d9a1.png)

**Comment by al-jabirr** [comment]
First of all I wish you well sir (leocg), you are supporting all of us despite being busy. thank you.When I plugged in a **USB 2.0** and found it worked fine on all ports. but **whenever I plugged in a 3.1 or type c It doesn't work. Only it's not working on my 3.1 and Type C flash drive or external hdd.** otherwise everything is fine Alhamdulillah.

**And I also tried using other usbmap.kext and usbinjectall.kext but didn't see any change.**

And I also think that I have the right USB mapping because I'm using your USB mapping (g5 5590) same to you sir, my laptop model is the same as your model. The problem lies elsewhere. Is it possible to fix it somehow sir?

![Screenshot 2022-12-03 at 8 49 13 PM](https://user-images.githubusercontent.com/8538407/205447357-660904ba-d62f-4264-8f57-040b5b5a9a5a.png)
![Screenshot 2022-12-03 at 8 54 22 PM](https://user-images.githubusercontent.com/8538407/205447367-93cc278f-20b7-46ee-bb48-0244fb4ae5b6.png)
